### DISCUSSION TITLE
ClamAV contrainer won't start anymore

### DISCUSSION THREAD

**Discussion by SageBilt** [nextcloud/all-in-one#2089] [Questions] [answered]
I have this issue where the ClamAV container won't start any more. I get a "Socket for clamd not found yet" error in the logs (see below). This issue only started after running "docker exec -e STOP_CONTAINERS=1 nextcloud-aio-mastercontainer /daily-backup.sh" and then "docker exec -e START_CONTAINERS=1 nextcloud-aio-mastercontainer /daily-backup.sh"

Currently I have had to remove ClamAV to get Nextcloud to start.

> Starting Freshclamd
> Starting ClamAV
> 
> Socket for clamd not found yet, retrying (0/90) ...ClamAV update process started at Thu Mar  2 14:04:19 2023
> daily.cld database is up-to-date (version: 26827, sigs: 2022011, f-level: 90, builder: raynman)
> main.cvd database is up-to-date (version: 62, sigs: 6647427, f-level: 90, builder: sigmgr)
> bytecode.cvd database is up-to-date (version: 334, sigs: 91, f-level: 90, builder: anvilleg)
> 
> Socket for clamd not found yet, retrying (1/90) ...
> Socket for clamd not found yet, retrying (2/90) ...
> Socket for clamd not found yet, retrying (3/90) ...
> Socket for clamd not found yet, retrying (4/90) ...
> Socket for clamd not found yet, retrying (5/90) ...
> Socket for clamd not found yet, retrying (6/90) ...
> Socket for clamd not found yet, retrying (7/90) ...
> Socket for clamd not found yet, retrying (8/90) ...
> Socket for clamd not found yet, retrying (9/90) ...
> Socket for clamd not found yet, retrying (10/90) ...
> Socket for clamd not found yet, retrying (11/90) ...
> Socket for clamd not found yet, retrying (12/90) ...
> Socket for clamd not found yet, retrying (13/90) ...
> Socket for clamd not found yet, retrying (14/90) ...
> Socket for clamd not found yet, retrying (15/90) ...
> Socket for clamd not found yet, retrying (16/90) ...
> Socket for clamd not found yet, retrying (17/90) ...
> Socket for clamd not found yet, retrying (18/90) ...
> Socket for clamd not found yet, retrying (19/90) ...
> Socket for clamd not found yet, retrying (20/90) ...
> Socket for clamd not found yet, retrying (21/90) ...
> Socket for clamd not found yet, retrying (22/90) ...
> Socket for clamd not found yet, retrying (23/90) ...
> Socket for clamd not found yet, retrying (24/90) ...
> Socket for clamd not found yet, retrying (25/90) ...
> Socket for clamd not found yet, retrying (26/90) ...
> Socket for clamd not found yet, retrying (27/90) ...
> Socket for clamd not found yet, retrying (28/90) ...
> Socket for clamd not found yet, retrying (29/90) ...
> Socket for clamd not found yet, retrying (30/90) ...
> Socket for clamd not found yet, retrying (31/90) ...
> Socket for clamd not found yet, retrying (32/90) ...
> Socket for clamd not found yet, retrying (33/90) ...
> Socket for clamd not found yet, retrying (34/90) ...
> Socket for clamd not found yet, retrying (35/90) ...
> Socket for clamd not found yet, retrying (36/90) ...
> Socket for clamd not found yet, retrying (37/90) ...
> Socket for clamd not found yet, retrying (38/90) ...
> Socket for clamd not found yet, retrying (39/90) ...
> Socket for clamd not found yet, retrying (40/90) ...
> Socket for clamd not found yet, retrying (41/90) ...
> Socket for clamd not found yet, retrying (42/90) ...
> Socket for clamd not found yet, retrying (43/90) ...
> Socket for clamd not found yet, retrying (44/90) ...
> Socket for clamd not found yet, retrying (45/90) ...
> Socket for clamd not found yet, retrying (46/90) ...
> Socket for clamd not found yet, retrying (47/90) ...
> Socket for clamd not found yet, retrying (48/90) ...
> Socket for clamd not found yet, retrying (49/90) ...
> Socket for clamd not found yet, retrying (50/90) ...
> Socket for clamd not found yet, retrying (51/90) ...
> Socket for clamd not found yet, retrying (52/90) ...
> Socket for clamd not found yet, retrying (53/90) ...
> Socket for clamd not found yet, retrying (54/90) ...
> Socket for clamd not found yet, retrying (55/90) ...
> Socket for clamd not found yet, retrying (56/90) ...
> Socket for clamd not found yet, retrying (57/90) ...
> Socket for clamd not found yet, retrying (58/90) ...
> Socket for clamd not found yet, retrying (59/90) ...
> Socket for clamd not found yet, retrying (60/90) ...
> Socket for clamd not found yet, retrying (61/90) ...
> Socket for clamd not found yet, retrying (62/90) ...
> Socket for clamd not found yet, retrying (63/90) ...
> Socket for clamd not found yet, retrying (64/90) ...
> Socket for clamd not found yet, retrying (65/90) ...
> Socket for clamd not found yet, retrying (66/90) ...
> Socket for clamd not found yet, retrying (67/90) ...
> Socket for clamd not found yet, retrying (68/90) ...
> Socket for clamd not found yet, retrying (69/90) ...
> Socket for clamd not found yet, retrying (70/90) ...
> Socket for clamd not found yet, retrying (71/90) ...
> Socket for clamd not found yet, retrying (72/90) ...
> Socket for clamd not found yet, retrying (73/90) ...
> Socket for clamd not found yet, retrying (74/90) ...
> Socket for clamd not found yet, retrying (75/90) ...
> Socket for clamd not found yet, retrying (76/90) ...
> Socket for clamd not found yet, retrying (77/90) ...
> Socket for clamd not found yet, retrying (78/90) ...
> Socket for clamd not found yet, retrying (79/90) ...
> Socket for clamd not found yet, retrying (80/90) ...
> Socket for clamd not found yet, retrying (81/90) ...
> Socket for clamd not found yet, retrying (82/90) ...
> Socket for clamd not found yet, retrying (83/90) ...
> Socket for clamd not found yet, retrying (84/90) ...
> Socket for clamd not found yet, retrying (85/90) ...
> Socket for clamd not found yet, retrying (86/90) ...
> Socket for clamd not found yet, retrying (87/90) ...
> Socket for clamd not found yet, retrying (88/90) ...
> Socket for clamd not found yet, retrying (89/90) ...
> Socket for clamd not found yet, retrying (90/90) ...
> Failed to start clamd
> Starting Freshclamd
> Starting ClamAV
> 
> Socket for clamd not found yet, retrying (0/90) ...ClamAV update process started at Thu Mar  2 14:05:53 2023
> daily.cld database is up-to-date (version: 26827, sigs: 2022011, f-level: 90, builder: raynman)
> main.cvd database is up-to-date (version: 62, sigs: 6647427, f-level: 90, builder: sigmgr)
> bytecode.cvd database is up-to-date (version: 334, sigs: 91, f-level: 90, builder: anvilleg)

**Comment by Zoey2936** [comment]
can confirm with normal daily update, but I think it is an upstream problem

**Reply by Zoey2936** [reply]
@szaimen do you have the same problem?

**Comment by szaimen** [comment]
I just tested and works for me: 

```
Starting Freshclamd
Starting ClamAV

or clamd not found yet, retrying (0/90) ...ClamAV update process started at Thu Mar  2 09:15:10 2023
daily.cld database is up-to-date (version: 26828, sigs: 2022427, f-level: 90, builder: raynman)
main.cvd database is up-to-date (version: 62, sigs: 6647427, f-level: 90, builder: sigmgr)
bytecode.cvd database is up-to-date (version: 334, sigs: 91, f-level: 90, builder: anvilleg)

or clamd not found yet, retrying (1/90) ...
or clamd not found yet, retrying (2/90) ...
or clamd not found yet, retrying (3/90) ...
or clamd not found yet, retrying (4/90) ...
or clamd not found yet, retrying (5/90) ...
or clamd not found yet, retrying (6/90) ...
or clamd not found yet, retrying (7/90) ...
or clamd not found yet, retrying (8/90) ...
or clamd not found yet, retrying (9/90) ...
or clamd not found yet, retrying (10/90) ...
or clamd not found yet, retrying (11/90) ...
or clamd not found yet, retrying (12/90) ...
or clamd not found yet, retrying (13/90) ...
or clamd not found yet, retrying (14/90) ...
or clamd not found yet, retrying (15/90) ...
or clamd not found yet, retrying (16/90) ...
or clamd not found yet, retrying (17/90) ...
or clamd not found yet, retrying (18/90) ...
or clamd not found yet, retrying (19/90) ...
or clamd not found yet, retrying (20/90) ...
or clamd not found yet, retrying (21/90) ...
or clamd not found yet, retrying (22/90) ...
or clamd not found yet, retrying (23/90) ...
or clamd not found yet, retrying (24/90) ...
or clamd not found yet, retrying (25/90) ...
or clamd not found yet, retrying (26/90) ...
or clamd not found yet, retrying (27/90) ...
or clamd not found yet, retrying (28/90) ...
or clamd not found yet, retrying (29/90) ...
or clamd not found yet, retrying (30/90) ...
or clamd not found yet, retrying (31/90) ...
or clamd not found yet, retrying (32/90) ...
or clamd not found yet, retrying (33/90) ...
or clamd not found yet, retrying (34/90) ...
or clamd not found yet, retrying (35/90) ...
or clamd not found yet, retrying (36/90) ...Thu Mar  2 09:15:46 2023 -> Limits: Global time limit set to 120000 milliseconds.
Thu Mar  2 09:15:46 2023 -> Limits: Global size limit set to 419430400 bytes.
Thu Mar  2 09:15:46 2023 -> Limits: File size limit set to 104857600 bytes.
Thu Mar  2 09:15:46 2023 -> Limits: Recursion level limit set to 17.
Thu Mar  2 09:15:46 2023 -> Limits: Files limit set to 10000.
Thu Mar  2 09:15:46 2023 -> Limits: MaxEmbeddedPE limit set to 41943040 bytes.
Thu Mar  2 09:15:46 2023 -> Limits: MaxHTMLNormalize limit set to 41943040 bytes.
Thu Mar  2 09:15:46 2023 -> Limits: MaxHTMLNoTags limit set to 8388608 bytes.
Thu Mar  2 09:15:46 2023 -> Limits: MaxScriptNormalize limit set to 20971520 bytes.
Thu Mar  2 09:15:46 2023 -> Limits: MaxZipTypeRcg limit set to 1048576 bytes.
Thu Mar  2 09:15:46 2023 -> Limits: MaxPartitions limit set to 50.
Thu Mar  2 09:15:46 2023 -> Limits: MaxIconsPE limit set to 100.
Thu Mar  2 09:15:46 2023 -> Limits: MaxRecHWP3 limit set to 16.
Thu Mar  2 09:15:46 2023 -> Limits: PCREMatchLimit limit set to 100000.
Thu Mar  2 09:15:46 2023 -> Limits: PCRERecMatchLimit limit set to 2000.
Thu Mar  2 09:15:46 2023 -> Limits: PCREMaxFileSize limit set to 104857600.
Thu Mar  2 09:15:46 2023 -> Archive support enabled.
Thu Mar  2 09:15:46 2023 -> AlertExceedsMax heuristic detection disabled.
Thu Mar  2 09:15:46 2023 -> Heuristic alerts enabled.
Thu Mar  2 09:15:46 2023 -> Portable Executable support enabled.
Thu Mar  2 09:15:46 2023 -> ELF support enabled.
Thu Mar  2 09:15:46 2023 -> Mail files support enabled.
Thu Mar  2 09:15:46 2023 -> OLE2 support enabled.
Thu Mar  2 09:15:46 2023 -> PDF support enabled.
Thu Mar  2 09:15:46 2023 -> SWF support enabled.
Thu Mar  2 09:15:46 2023 -> HTML support enabled.
Thu Mar  2 09:15:46 2023 -> XMLDOCS support enabled.
Thu Mar  2 09:15:46 2023 -> HWP3 support enabled.
Thu Mar  2 09:15:46 2023 -> Self checking every 600 seconds.
Thu Mar  2 09:15:46 2023 -> Set stacksize to 1048576
socket found, clamd started.


```

**Comment by szaimen** [comment, marked as answer]
However will improve this with https://github.com/nextcloud/all-in-one/pull/2092

**Reply by szaimen** [reply]
This is now released with v4.5.1 Beta. Testing and feedback is welcome! See https://github.com/nextcloud/all-in-one#how-to-switch-the-channel

**Comment by Zoey2936** [comment]
works again for me, seems like a temporary problem from the clamav database

**Reply by szaimen** [reply]
> works again for me, seems like a temporary problem from the clamav database

likely their service was down for a while...

**Reply by SageBilt** [reply]
Yes it works now.
Thanks you!
